File: 2022/01/16/main.go
package main

func maxDistToClosest(seats []int) int {
	d, e := 0, 0
	for i := 0; i < len(seats); i++ {
		if e == i {
			d = e
		} else {
			d = max(d, (e+e%2)/2)
		}
		if seats[i] == 1 {
			e = 0
		} else {
			e++
		}
	}

	return max(d, e)
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

File: 2022/01/16/main.py
class Solution:
    def maxDistToClosest(self, seats: List[int]) -> int:
        d, e = 0, 0
        for i in range(len(seats)):
            if e == i:
                d = e
            else:
                d = max(d, (e+e%2)/2)
            if seats[i] == 1:
                e = 0
            else:
                e += 1

        return max(d, e)